python/cudf/cudf/_lib/join.pyx

```diff
@@ -222,3 +201,20 @@ cpdef join(Table lhs,
         index = None
     data_ordered_dict = OrderedDict(zip(result_col_names, all_cols_py))
     return Table(data=data_ordered_dict, index=index)
+
+
+def compute_result_col_names(lhs, rhs, how):
+    """
+    Determine the names of the data columns in the result of
+    a libcudf join, based on the original left and right frames
+    as well as the type of join that was performed.
+    """
+    if how in ("left", "inner", "outer", "leftsemi", "leftanti"):
+        a = list(lhs._data.keys())
+        b = [] if how not in ("left", "inner", "outer") else \
+            [k for k in rhs._data.keys() if k not in lhs._data.keys()]
+        return a+b
```

Comment:
> Alternatively could use `itertools.chain` to make it read more nicely / clearly.

@kkraus14, for my own education.  If the goal is to rid the `*`, like this?

```
    a = lhs._data.keys()
    if how in {"left", "inner", "outer"}:
        return list(chain(a, [k for k in rhs..... if k not in lhs.....]))
    return list(a)
```

Reply:
Something about this isn't quite right - looking into it now.

Reply:
Fixed

Reply:
@philtrade you actually wouldn't want to materialize that inner list since it's wasteful in creating a Python list object. Instead you'd want to chain the `dict_keys` iterator in `a` with the generator expression you're using to build that inner list.

Using the `*` will unpack the generator expression, but it's a bit hard to read / follow where using `itertools.chain` makes the intent much clearer.

Reply:
> with the generator expression you're using to build that inner list.

I see!   Using `chain()` to save the `*` part I did get, but I didn't know substituting `[..]` of list comprehension with `(...)` would make it a generator, LOL.  Thanks to pointing that out!